### PR TITLE
feat: Events api reconcilation

### LINKED ISSUE #1304: Events API Reconciliation
### Description
There are quite a few event types that are missing from [inner_events.go](https://github.com/slack-go/slack/blob/master/slackevents/inner_events.go). It would be great if someone could go through and reconcile the list and add any missing events.

### (Optional) Slack's documentation
https://api.slack.com/events?filter=Events

### PR DESCRIPTION
**Pull request by Aryakoste**
Fixes https://github.com/slack-go/slack/issues/1304

### CODE REVIEW

**Comment by Aryakoste** [comment]
I will go through and check if I have added all the events. Just let me know if the additional events that I have added is done correctly or not. @lorenzoaiello

File: slackevents/inner_events.go [hunk 1]
```diff
@@ -649,6 +649,229 @@ type SharedInvite struct {
 	IsExternalLimited bool            `json:"is_external_limited,omitempty"`
 }
 
+type AppRateLimitedEvent struct {
+	Type       string `json:"type"`
+	Token      string `json:"token"`
+	TeamID     string `json:"team_id"`
+	MinuteRate int    `json:"minute_rate"`
```
Comment (lorenzoaiello) [inline slackevents/inner_events.go]: I believe this might need to be `json:"minute_rate_limited"`? 

ref: https://api.slack.com/events/app_rate_limited

Comment (lorenzoaiello) [reply slackevents/inner_events.go]: There is also a test missing for this.

File: slackevents/inner_events.go [hunk 2]
```diff
@@ -649,6 +649,229 @@ type SharedInvite struct {
...
+}
+
+type DndUpdatedUserEvent struct {
+	Type      string `json:"type"`
+	User      string `json:"user"`
+	DndStatus struct {
+		DndEnabled     bool  `json:"dnd_enabled"`
+		NextDndStartTs int64 `json:"next_dnd_start_ts"`
+		NextDndEndTs   int64 `json:"next_dnd_end_ts"`
+		SnoozeEnabled  bool  `json:"snooze_enabled"`
```
Comment (lorenzoaiello) [inline slackevents/inner_events.go]: `snooze_enabled` and `snooze_enabled` do not appear to be documented fields. Did they show up in testing or is this just a copy/paste?

ref: https://api.slack.com/events/dnd_updated_user

Comment (lorenzoaiello) [reply slackevents/inner_events.go]: There is also a test missing for this.

File: slackevents/inner_events.go [hunk 3]
```diff
@@ -649,6 +649,229 @@ type SharedInvite struct {
...
+	} `json:"dnd_status"`
+}
+
+type EmailDomainChangedEvent struct {
+	Type        string `json:"type"`
+	EmailDomain string `json:"email_domain"`
+	EventTs     string `json:"event_ts"`
+}
+
+type FileCommentAddedEvent struct {
```
Comment (lorenzoaiello) [inline slackevents/inner_events.go]: I appreciate adding this but the documentation indicates this is no longer served.

ref: https://api.slack.com/events/file_comment_added

File: slackevents/inner_events.go [hunk 4]
```diff
@@ -649,6 +649,229 @@ type SharedInvite struct {
...
+		Timestamp int64  `json:"timestamp"`
+		User      string `json:"user"`
+		Comment   string `json:"comment"`
+	} `json:"comment"`
+	File struct {
+		ID string `json:"id"`
+	} `json:"file"`
+}
+
+type FileCommentDeletedEvent struct {
```
Comment (lorenzoaiello) [inline slackevents/inner_events.go]: I appreciate adding this but the documentation indicates this is no longer served.

ref: https://api.slack.com/events/file_comment_deleted

File: slackevents/inner_events.go [hunk 5]
```diff
@@ -649,6 +649,229 @@ type SharedInvite struct {
...
+	Type    string `json:"type"`
+	Comment struct {
+		ID string `json:"id"`
+	} `json:"comment"`
+	File struct {
+		ID string `json:"id"`
+	} `json:"file"`
+}
+
+type FileCommentEditedEvent struct {
```
Comment (lorenzoaiello) [inline slackevents/inner_events.go]: I appreciate adding this but the documentation indicates this is no longer served.

ref: https://api.slack.com/events/file_comment_edited

File: slackevents/inner_events.go [hunk 6]
```diff
@@ -649,6 +649,229 @@ type SharedInvite struct {
...
+	} `json:"file"`
+}
+
+type GroupCloseEvent struct {
+	Type    string `json:"type"`
+	User    string `json:"user"`
+	Channel string `json:"channel"`
+}
+
+type GroupHistoryChangedEvent struct {
```
Comment (lorenzoaiello) [inline slackevents/inner_events.go:698]: There is also a test missing for this.

File: slackevents/inner_events.go [hunk 7]
```diff
@@ -649,6 +649,229 @@ type SharedInvite struct {
...
+	Type    string `json:"type"`
+	Latest  string `json:"latest"`
+	Ts      string `json:"ts"`
+	EventTs string `json:"event_ts"`
+}
+
+type GroupOpenEvent struct {
```
Comment (lorenzoaiello) [inline slackevents/inner_events.go:705]: There is also a test missing for this.

File: slackevents/inner_events.go [hunk 8]
```diff
@@ -649,6 +649,229 @@ type SharedInvite struct {
...
+
+type SubteamCreatedEvent struct {
+	Type    string  `json:"type"`
+	Subteam SubTeam `json:"subteam"`
+}
+
+type SubteamMembersChangedEvent struct {
+	Type         string   `json:"type"`
+	SubteamID    string   `json:"subteam_id"`
+	TeamID       string   `json:"team_id"`
```
Comment (lorenzoaiello) [inline slackevents/inner_events.go]: I think you are missing the `date_previous_update`, `date_update`, `added_users_count`, and `removed_users_count` fields.

ref: https://api.slack.com/events/subteam_members_changed

File: slackevents/inner_events.go [hunk 9]
```diff
@@ -649,6 +649,229 @@ type SharedInvite struct {
...
+
+type SubteamUpdatedEvent struct {
+	Type    string  `json:"type"`
+	Subteam SubTeam `json:"subteam"`
+}
+
+type TeamDomainChangeEvent struct {
+	Type   string `json:"type"`
+	URL    string `json:"url"`
+	Domain string `json:"domain"`
```
Comment (lorenzoaiello) [inline slackevents/inner_events.go:796]: I think you are missing the `team_id` field.

ref: https://api.slack.com/events/team_domain_change

File: slackevents/inner_events.go [hunk 10]
```diff
@@ -649,6 +649,229 @@ type SharedInvite struct {
...
+}
+
+type TeamRenameEvent struct {
+	Type string `json:"type"`
+	Name string `json:"name"`
```
Comment (lorenzoaiello) [inline slackevents/inner_events.go]: I think you are missing the `team_id` field.

ref: https://api.slack.com/events/team_rename

File: slackevents/inner_events.go [hunk 11]
```diff
@@ -649,6 +649,229 @@ type SharedInvite struct {
...
+}
+
+type UserChangeEvent struct {
+	Type string `json:"type"`
+	User struct {
```
Comment (lorenzoaiello) [inline slackevents/inner_events.go]: Please double check the struct, there are quite some fields missing.

ref: https://api.slack.com/events/user_change

File: slackevents/inner_events.go [hunk 12]
```diff
@@ -738,6 +961,52 @@ const (
 	TeamAccessRevoked = EventsAPIType("team_access_revoked")
 	// UserProfileChanged is sent if a user's profile information has changed.
 	UserProfileChanged = EventsAPIType("user_profile_changed")
+	// ChannelHistoryChanged The history of a channel changed
```
Comment (lorenzoaiello) [inline slackevents/inner_events.go:1197]: I believe `AppRateLimited` is missing here

**Review by lorenzoaiello — COMMENTED** [review]
Hi @Aryakoste ,

Apologies for the delay in getting a review on this. It's close and adds so many new events -- thank you for your work! I've left some comments.

File: slackevents/inner_events.go (hunk 2 above)
Comment (Aryakoste) [reply slackevents/inner_events.go]: Sorry i used DndUpdatedEvent struct same for DndUpdatedUserEvent forgot to remove the snooze property. Will change this and add test. Thanks.

File: slackevents/inner_events.go (hunk 12 above)
Comment (Aryakoste) [reply slackevents/inner_events.go:1197]: The AppRateLimited event is already added in outer_events.go so wasnt able to replicate it here. I will remove this event from here.

**Comment by Aryakoste** [comment]
@lorenzoaiello  I have added all events with tests and made changes.

**Review by lorenzoaiello — APPROVED** [review]
Thanks for the contribution and updates @Aryakoste ! It's much appreciated.

**Pull request merged**